Implemented SchoolController with tests, added Mockito test dependecy

Cosimo5lli opened the pull request:
Closes #12

Cosimo5lli committed:
Implemented SchoolController with tests, added Mockito test dependecy

Closes #12

Cosimo5lli committed:
Merge remote-tracking branch 'origin/master' into SchoolController-Issue-12

LorenzoBettini commented on com.examples.exercise/pom.xml:
```diff
@@ -62,6 +62,12 @@
+		<dependency>
+			<groupId>org.mockito</groupId>
+			<artifactId>mockito-core</artifactId>
+			<version>2.2.15</version>
```
use the version of mockito we've always used in class: 1.10.19

Cosimo5lli replied:
@LorenzoBettini i've updated the branch with this modification

LorenzoBettini commented on com.examples.exercise/src/test/java/com/examples/exercise/SchoolControllerTest.java:
```diff
@@ -0,0 +1,39 @@
+	public void testGetAllStudentsWhenThereIsOneStudent() {
+		students.add(new Student());
+		List<Student> allStudents = schoolController.getAllStudents();
+		assertEquals(1, allStudents.size());	
```
it's not completely required, by I would have expected a verify also in this test

Cosimo5lli replied:
@LorenzoBettini i've also added the verify

Cosimo5lli committed:
Modified Mockito version, added verify statement to test

Mockito version changed from 2.2.15 to 1.10.19
Added verify statement in testGetAllStudentsWhenThereAreNoStudents()

Cosimo5lli committed:
Merge remote-tracking branch 'origin/master' into SchoolController-Issue-12

LorenzoBettini commented on com.examples.exercise/src/main/java/com/examples/exercise/Database.java:
```diff
@@ -0,0 +1,10 @@
+public interface Database {
+
+	List<Student> getAllStudentsList();
+
```
There are still additional new lines in this interface

LorenzoBettini commented on com.examples.exercise/src/main/java/com/examples/exercise/SchoolController.java:
```diff
@@ -0,0 +1,20 @@
+
+public class SchoolController {
+	private Database database;
+	
```
there are still additional newlines in this class

LorenzoBettini commented:
besides my last comments, you should now merge with master, since another pull request has been merged (https://github.com/LorenzoBettini/example.exercise/pull/19). You'll get conflicts and you should merge your modifications, without removing what's already been implemented in the other pull request

Cosimo5lli committed:
Merge remote-tracking branch 'origin/master' into
SchoolController-Issue-12

Conflicts:
	com.examples.exercise/pom.xml
	com.examples.exercise/src/main/java/com/examples/exercise/Database.java

	Re-added scope test in pom.xml for mockito

Cosimo5lli committed:
Removed unused import in SchoolController

LorenzoBettini commented on com.examples.exercise/src/test/java/com/examples/exercise/SchoolControllerTest.java:
```diff
@@ -0,0 +1,40 @@
+		verify(database, times(1)).getAllStudentsList();
+		assertEquals(1, allStudents.size());	
+	}
+
```
During merging, you removed the tests of https://github.com/LorenzoBettini/example.exercise/pull/19/files

Cosimo5lli replied:
@LorenzoBettini I don't know what to do, they are present in my local copy of the branch, I can see and modify them from eclipse and from filesystem. In addition "git diff origin/master HEAD" only show I added SchoolController.java and SchoolControllerTest.java and modified pom.xml

LorenzoBettini replied:
Look at the differences here: https://github.com/LorenzoBettini/example.exercise/pull/18/files and you see that the other methods are not there

Cosimo5lli replied:
But shuldn't this view show only files changed in this commit? The tests you referenced are in other source file (StudentControllerTest.java) not touched at all by my commit nor by my merge and consequent conflicts correction. I'm sorry, even if this is not the case and file have effectively been deleted I don't know how to restore them in this branch, since they exists in my local copy and git says I'm up to date

LorenzoBettini replied:
SORRY!!! I was confusing SchoolController and StudentController!!!

LorenzoBettini commented on com.examples.exercise/src/main/java/com/examples/exercise/SchoolController.java:
```diff
@@ -0,0 +1,17 @@
+	public List<Student> getAllStudents() {
+		return database.getAllStudentsList();
+	}
+
```
During merging, you removed the implementation of update of https://github.com/LorenzoBettini/example.exercise/pull/19/files

Cosimo5lli replied:
@LorenzoBettini I don't know what to do, they are present in my local copy of the branch, I can see and modify them from eclipse and from filesystem. In addition "git diff origin/master HEAD" only show I  added SchoolController.java and SchoolControllerTest.java and modified pom.xml
